### DISCUSSION TITLE
Coupling with reaktoro

### DISCUSSION THREAD

**Discussion by MagunRa** [pybamm-team/PyBaMM#4138] [Q&A]
Hello,
I want to build a model for an aqueos system which is sensitive for the pH. I am assuming that the species in the electrolye reache equillibirum faster than the electro chemical reactions, I want to use [reaktoro](https://reaktoro.org/index.html) to get the equillibrium conditions.  This means I have a dc/dt from the electrochemical reaction and a "post step" processing like c_eq=f(c_n) where c_eq is the "real result" and starting condition for the next step.

I am aware of this [discussion](https://github.com/pybamm-team/PyBaMM/discussions/3206), but I want to let the solver choose appropiate time stepping, since the relevant time scales may be changing during a simulation. Therefore doing step with a fixed dt seems not optimal.

Can someone give me a direction how I can do this "post step" ?
Best Regards

**Comment by brosaplanella** [comment]
The step method actually does adaptive time-stepping, it just stops after a `dt`.
